pkg/steps/source.go

```diff
@@ -539,20 +542,80 @@ func hintsAtInfraReason(logSnippet string) bool {
 		strings.Contains(logSnippet, "connection reset by peer")
 }
 
-func waitForBuildOrTimeout(ctx context.Context, buildClient BuildClient, namespace, name string) error {
-	return waitForBuild(ctx, buildClient, namespace, name)
-}
-
-func waitForBuild(ctx context.Context, buildClient BuildClient, namespace, name string) error {
+func waitForBuildOrTimeout(
```

Comment:
`waitForBuildOrTimeout` - I would expect from the extra `OrTimeout` suffix that this function is the same as `waitForBuild` but I accepts an extra `timeout time.Duration` argument. Is it reasonable?

Reply:
Note that I'm not changing the name, as the `diff` output can seem to indicate.  What is funny is the original implementation _did not_ have a timeout.  It just waited forever, despite its name.

We now _do_ have a timeout parameter, only it's passed implicitly via the `podClient` (later retrieved by its `GetPodPendingTimeout` method).  I thought about making it explicit, but it didn't seem necessary since we can get it directly from the client object.